apps/legacy/src/pages/Swap/Swap.tsx

```diff
@@ -254,6 +254,11 @@ export function Swap() {
     }
   }
 
+  const showEngineNotice = useMemo(
+    () => quote && (isParaswapQuote(quote) || isJupiterQuote(quote)),
+    [quote],
+  );
```

Comment:
IMO there's no need to use memoization here. 
The overhead useMemo introduces cost much more than 3 simple comparisons.

```suggestion
  const showEngineNotice = quote && (isParaswapQuote(quote) || isJupiterQuote(quote));
  );
```